pkg/services/ngalert/notifier/channels/telegram.go

```diff
@@ -168,15 +168,28 @@ func (tn *TelegramNotifier) Notify(ctx context.Context, as ...*types.Alert) (boo
 
 func (tn *TelegramNotifier) buildTelegramMessage(ctx context.Context, as []*types.Alert) (map[string]string, error) {
 	var tmplErr error
-	defer func() {
-		if tmplErr != nil {
-			tn.log.Warn("failed to template Telegram message", "error", tmplErr)
-		}
-	}()
-
 	tmpl, _ := TmplText(ctx, tn.tmpl, as, tn.log, &tmplErr)
+	messageText := tmpl(tn.settings.Message)
+	if tmplErr != nil {
+		tn.log.Warn("Failed to template Telegram message", "error", tmplErr, "template", tn.settings.Message)
+		tmplErr = nil
+	}
```

Comment:
I think this pattern we should adopt for all template expansion rather than blanket log message at the end of the notification processing